### DISCUSSION TITLE
More productive use of mentions for PRs and issues

### DISCUSSION THREAD

**Discussion by Tokazama** [JuliaLang/julia#46536] [General]
I think Julia could do a better job of managing and inviting meaningful contributions from non-core developers. As someone who is not a "core" contributor, once a PR is ready for review it isn't clear when or if it will be looked at. Sometimes I see people just spamming core devs in hopes that it moves things along. I assume devs are already answering as many mentions as they can and may occasionally have a chance to flick through everything else, seeing if the title stands out as super important. That's why I try to avoid mentioning anyone unless there's an actual reason they should personally be involved. However, it's not always clear who is the most appropriate person to get involved.

With that in mind, would it be helpful for developers if there was a list of topics/methods that specific developers names are attached to so that they are they immediately notified instead of everyone being spammed or issues/PRs never being addressed?

**Comment by DilumAluthge** [comment]
> if there was a list of topics/methods that specific developers names are attached to so that they are they immediately notified

I think using the GitHub CODEOWNERS file is going to be the easiest way to implement this. Then, when someone opens a PR, GitHub automatically requests a review from the code owners of the files modified by the PR.

**Reply by Tokazama** [reply]
I see that you're the only one on the CODEOWNERS file right now. Is this just because it's relatively new or is this not a useful strategy for developers. I want to do whatever will be the most productive for everyone, so if this is off-putting for some devs I don't want to be pushy

**Reply by LilithHafner** [reply]
I'd like to be notified whenever someone opens a PR that modifies `base/sort.jl` but I didn't want to "own" the code in any way. Is CODEOWNERS appropriate for this?

**Comment by Tokazama** [comment]
Cool! I didn't even know that existed. I think we need to split up some files if we want to rely on this. Hundreds/thousands of lines of code per file might just end up notifying too many people.
